alluxio/alluxio.sh

```diff
@@ -0,0 +1,177 @@
+      download_file "${file}"
+      mv "${filename}" "${ALLUXIO_HOME}/conf/${filename}"
+    done
+  fi
+
+  # Configure client applications
+  mkdir -p "${SPARK_HOME}/jars/"
+  ln -s "${ALLUXIO_HOME}/client/alluxio-client.jar" "${SPARK_HOME}/jars/alluxio-client.jar"
+  mkdir -p "${HIVE_HOME}/lib/"
+  ln -s "${ALLUXIO_HOME}/client/alluxio-client.jar" "${HIVE_HOME}/lib/alluxio-client.jar"
+  mkdir -p "${HADOOP_HOME}/lib/"
+  ln -s "${ALLUXIO_HOME}/client/alluxio-client.jar" "${HADOOP_HOME}/lib/alluxio-client.jar"
+  if [[ "${ROLE}" == "Master" ]]; then
+    systemctl restart hive-metastore hive-server2
+  fi
+
+  # Optionally configure Presto
+  # OK to fail in this section
+  set +o errexit
+  PRESTO_HOME=${PRESTO_HOME:-$(ls -d -- /presto-server-*)}
+  if [[ -n $PRESTO_HOME ]]; then
+    mkdir -p "${PRESTO_HOME}/plugin/hive-hadoop2/"
+    ln -s "${ALLUXIO_HOME}/client/alluxio-client.jar" "${PRESTO_HOME}/plugin/hive-hadoop2/alluxio-client.jar"
+    systemctl restart presto
+  fi
+  set -o errexit # errors not ok anymore
+
+  ln -s "${ALLUXIO_HOME}/bin/alluxio" /usr/bin
+}
+
+# Configure alluxio-site.properties
+function configure_alluxio() {
+  cp "${ALLUXIO_HOME}/conf/alluxio-site.properties.template" ${ALLUXIO_SITE_PROPERTIES}
+
+  append_alluxio_property alluxio.master.hostname "${MASTER_FQDN}"
+
+  local root_ufs_uri
+  root_ufs_uri=$(/usr/share/google/get_metadata_value attributes/alluxio_root_ufs_uri)
+  append_alluxio_property alluxio.master.mount.table.root.ufs "${root_ufs_uri}"
+
+  local mem_size
+  mem_size=$(get_default_mem_size)
+  append_alluxio_property alluxio.worker.memory.size "${mem_size}"
+  append_alluxio_property alluxio.worker.tieredstore.level0.alias "MEM"
+  append_alluxio_property alluxio.worker.tieredstore.level0.dirs.path "/mnt/ramdisk"
+  append_alluxio_property alluxio.worker.tieredstore.levels "1"
+
+  append_alluxio_property alluxio.master.security.impersonation.root.users "*"
+  append_alluxio_property alluxio.master.security.impersonation.root.groups "*"
+  append_alluxio_property alluxio.master.security.impersonation.client.users "*"
+  append_alluxio_property alluxio.master.security.impersonation.client.groups "*"
+  append_alluxio_property alluxio.security.login.impersonation.username "none"
+  append_alluxio_property alluxio.security.authorization.permission.enabled "false"
+
+  local site_properties
+  site_properties=$(/usr/share/google/get_metadata_value attributes/alluxio_site_properties || true)
+  local property_delimiter=";"
+  if [[ "${site_properties}" ]]; then
+    IFS="${property_delimiter}" read -ra conf <<<"${site_properties}"
+    for property in "${conf[@]}"; do
+      local key=${property%%"="*}
+      local value=${property#*"="}
+      append_alluxio_property "${key}" "${value}"
+    done
+  fi
+}
+
+# Start the Alluxio server process
+function start_alluxio() {
+  if [[ "${ROLE}" == "Master" ]]; then
+    alluxio formatMaster
+    ${ALLUXIO_HOME}/bin/alluxio-start.sh master
```

Comment:
Instead of starting Alluxio daemons manually, we need to create SystemD services for Alluxio, so it will be started after VM restart.

Here is an [example](https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/blob/fc85acf6b17b3cc4a00ce8bb091940276844f6be/beam/beam.sh#L87:L105)

Reply:
done